website/docs/tutorials/08-interacting-with-starknet/scripting.md

```diff
@@ -0,0 +1,194 @@
+# Scripting
+In order to automate a process of your desire that includes protostar operations, you may want to assemble some protostar commands and put them in a single script.
+
+This tutorial shows a simple example of how to do such a thing using scripting in bash.
+
+### File Structure
+
+Let's create the following file structure:
+
+```
+- src
+  - basic_contract.cairo
+- tests
+  - test_basic_contract.cairo
+- protostar.toml
+```
+
+#### Basic contract
+
+```cairo
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+
+@storage_var
+func balance() -> (res: felt) {
+}
+
+@constructor
+func constructor{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    initial_balance: felt
+) {
+    balance.write(initial_balance);
+    return ();
+}
+
+
+@external
+func increase_balance{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    amount: felt
+) {
+    let (res) = balance.read();
+    balance.write(res + amount);
+    return ();
+}
+
+@view
+func get_balance{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+    let (res) = balance.read();
+    return (res,);
+}
+```
+
+#### Test for the contract
+
+```cairo
+%lang starknet
+
+from starkware.cairo.common.math import assert_le
+
+@contract_interface
+namespace BasicContract {
+  func increase_balance(amount: felt) -> () {
+  }
+  func get_balance() -> (res: felt) {
+  }
+}
+
+@external
+func setup_basic() {
+    %{
+      context.contract_address = deploy_contract("./src/basic_contract.cairo", [100]).contract_address
+    %}
+    return ();
+}
+
+@external
+func test_basic{syscall_ptr: felt*, range_check_ptr}() {
+    alloc_locals;
+
+    local contract_address: felt;
+    %{ ids.contract_address = context.contract_address %}
+
+    let (res,) = BasicContract.get_balance(contract_address=contract_address);
+    assert res = 100;
+
+    BasicContract.increase_balance(contract_address=contract_address, amount=50);
+    let (res,) = BasicContract.get_balance(contract_address=contract_address);
+    assert res = 150;
+
+    return ();
+}
+```
+
+#### protostar.toml file
```

Comment:
Can we remove this section, it is not relevant imo.

Reply:
I think it is relevant. Eventually, we want the commands in the script to be as simple as possible, we don't want something like this:
```
protostar declare ./build/main.json --gateway-url http://127.0.0.1:5050/ --account-address 0x0668c707Af664985EDfB3Ea616a0189660C2C54ADCa787B870B55DC314b91d94 --private-key-path priv.key --max-fee auto --chain-id 1536727068981429685321
```
Instead, I believe it is better and cleaner to have something like this:
```
protostar declare ./build/main.json
```
I think it is worth mentioning why all commands are so thin.

Reply:
Ok I agree